base/iobuffer.jl

```diff
@@ -92,6 +92,13 @@ end
     return byte
 end
 
+@inline function unsafe_read(from::AbstractIOBuffer, ::Type{UInt8})
```

Comment:
Thinking more, I don't think we'd want to do the `@boundscheck` for `read`, imagine the following two cases:

```julia
@inbounds for i = 1:n
    b = read(io, UInt8)
    A[i] = f(b)
end
```
or
```julia
@inbounds begin
    b = read(io, UInt8)
    A[Int(b)] = b * 2
end
```
I think it'd be too subtle for users if they ever happen to be using read in a block where they're doing `@inbounds`.